ToolBox/PsychSignalDataConverter/PsychSignalDataConverter.cs

```diff
@@ -0,0 +1,157 @@
+                if (!DateTime.TryParse(csv[2], out timestamp))
+                {
+                    Log.Error($"Failed to parse timestamp: {csv[2]}");
+                    continue;
+                }
+
+                var dataFilePath = Path.Combine(sentimentFolder, ticker, $"{timestamp:yyyyMMdd}.csv");
+
+                // Avoids having to re-open the file every time we want to write to it
+                TickerData handler;
+                if (!tickerFileHandlers.TryGetValue(ticker, out handler))
+                {
+                    if (!tickerFolders.Contains(ticker))
+                    {
+                        Directory.CreateDirectory(Path.Combine(sentimentFolder, ticker));
+                        tickerFolders.Add(ticker);
+                    }
+                    tickerFileHandlers[ticker] = new TickerData(dataFilePath);
+                    handler = tickerFileHandlers[ticker];
+                }
+                if (handler.DataPath != dataFilePath)
+                {
+                    tickerFileHandlers[ticker].UpdateWriter(dataFilePath);
+                }
+                // We need to flush the previous data if our ticker has changed in order to completely write all data to disk.
+                // Previously, the last day's data would not be written because the file was never closed. 
+                if (previousTicker != ticker && !string.IsNullOrEmpty(previousTicker))
+                {
+                    tickerFileHandlers[previousTicker].Writer.Close();
+                    tickerFileHandlers.Remove(previousTicker);
+                }
+
+                // SOURCE[0],SYMBOL[1],TIMESTAMP_UTC[2],BULLISH_INTENSITY[3],BEARISH_INTENSITY[4],BULL_MINUS_BEAR[5],BULL_SCORED_MESSAGES[6],BEAR_SCORED_MESSAGES[7],BULL_BEAR_MSG_RATIO[8],TOTAL_SCANNED_MESSAGES[9]
+                handler.Writer.WriteLine(ToCsv(timestamp, csv.Skip(3)));
+                previousTicker = ticker;
+            }
+            
+            // Free final writer so that we don't get an IOException due to the final file being open
+            tickerFileHandlers[previousTicker].Writer.Close();
+
+            foreach (var tickerDataFolder in Directory.GetDirectories(sentimentFolder))
+            {
+                Compression.ZipDirectory(tickerDataFolder, $"{tickerDataFolder}.zip", false);
+                Directory.Delete(tickerDataFolder, true);
+            }
+        }
+
+        /// <summary>
+        /// Converts line of psychsignal data to LEAN's csv format
+        /// </summary>
+        /// <param name="timestamp">Timestamp as a string to use for filename</param>
+        /// <param name="csvData">Data as it comes from data vendor</param>
+        /// <returns></returns>
+        public string ToCsv(DateTime timestamp, IEnumerable<string> csvData)
+        {
+            return $"{timestamp.Subtract(new DateTime(timestamp.Year, timestamp.Month, timestamp.Day)).TotalMilliseconds},{string.Join(",", csvData)}";
```

Comment:
nit - if you're looking for the total milliseconds since midnight of a `DateTime` instance, you can use `timestamp.TimeOfDay.TotalMilliseconds` instead of needing to perform the subtraction